### PR TITLE
persist: add test coverage of concurrent usage of PostgresConsensus

### PR DESCRIPTION
**Pull request by danhhz**
In particular, including open. This was a failed attempt to repro #12560, but
it seems useful to have on its own.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

**Comment by aljoscha** [comment]
> This was an attempt to repro https://github.com/MaterializeInc/materialize/issues/12560, but
> it seems useful to have on it's own.

Was the attempt successful?

File: src/persist/src/postgres.rs [hunk 1]
```diff
@@ -364,4 +367,88 @@ mod tests {
...
+                Ok(x) => return x,
+                Err(err) => {
+                    info!("external operation {} failed, retrying in: {}", name, err);
+                }
+            }
+        }
+    }
+
+    // A failed attempt to reproduce #12560, but it seems useful on it's own.
+    // Perhaps we should make a `consensus_concurrency_test` so we can have this
```
Comment (ruchirK) [inline src/persist/src/postgres.rs:387]: that seems useful to me!

**Review by ruchirK — APPROVED** [review]
LGTM!

**Comment by benesch** [comment]
@danhhz did you mean to keep this open?

**Comment by danhhz** [comment]
Yup, didn't mean to close it :)

**Pull request closed without merging**